.github/pull_request_template.md

```diff
@@ -0,0 +1,84 @@
+<!--
+You've got a Pull Request you want to submit? Awesome!
+This PR template is here to help ensure you're setup for success:
+  please fill it out to help ensure that your PR is complete and ready for approval.
+-->
+
+**JIRA Ticket:**
+<https://jira.cms.gov/browse/xxxx>
+
+**User Story or Bug Summary:**
+<!-- Please copy-paste the brief user story or bug description that this PR is intended to address. -->
+
+
+### What Does This PR Do?
+
+<!--
+Add detailed discussion of changes here
+This is likely a summary, or the complete contents, of your commit messages.
+-->
+
+
+### What Should Reviewers Watch For?
+
+If you're reviewing this PR, please check these things, in particular:
+
+* TODO
+
+<!--
+Add some items to the list above, or remove the entire section if it doesn't apply for some reason.
+
+Common items include:
+* Is this likely to address the goals expressed in the user story?
+* Are any additional documentation updates needed?
+* Are there any unhandled and/or untested edge cases you can think of?
+* Is user input properly sanitized & handled?
+* Does this make any backwards-incompatible changes that might break end user clients?
+* Can you find any bugs if you run the code locally and test it manually?
+-->
+
+
+### What Security Implications Does This PR Have?
+
+Submitters should complete the following questionnaire:
+
+* If the answer to any of these questions is **Yes**, then here's a link to the associated Security Impact Assessment: N/A.
+    * Does this PR add any new software dependencies? **Yes** or **No**.
+    * Does this PR modify or invalidate any of our security controls? **Yes** or **No**.
+    * Does this PR store or transmit data that was not stored or transmitted before? **Yes** or **No**.
+* If the answer to any of these questions is **Yes**, then please add @StewGoin as a reviewer, and note that this PR should not be merged unless/until he also approves it.
+    * Do you think this PR requires additional review of its security implications for other reasons? **Yes** or **No**.
+
+
+### What Needs to Be Deployed Before this PR?
+
+This PR cannot be deployed until the following pre-requisite changes have been fully deployed:
+
+* CMSgov/some_repo#42
+
+<!--
+Add some items to the list above, or remove the entire section if it doesn't apply.
+
+Common items include:
+* Database migrations (which should always go out in PRs by themselves, to reduce risk).
+* New features in external dependencies (e.g. BFD).
+-->
+
+
+### Submitter Checklist
+
+Before requesting final review, I've gone through and verified that this PR complies with the following requirements:
+
+* [ ] I've verified that this PR is as small as it can reasonably be, in order to:
+    1. Ensure that any problems it causes have a small "blast radius".
+    2. Ensure that it'll be easier to rollback if that becomes necessary.
+    3. Ease the burden on reviewers.
+* [ ] I've verified that the individual commits in this PR all:
```

Comment:
Does this matter?  If we are using "Squash and Merge", then this won't matter, right?  
I'm worried that this will be hard to follow, specifically in situations where a heavy dose of troubleshooting is occurring, and sometimes the commit comments are simply "fixing" or "reverting" (for example).

Reply:
I think we have an item in the Engineering Standards right now asking folks to _not_ squash.

Personally, I'm a big fan of a rich & useful history. I **like** knowing how the code evolved as it was drafted: that this line had a bug that was fixed, that that other function was renamed to be more clear, etc. I also think it's good just to be honest about how code doesn't just pop into the world fully-formed and beautiful.

However. That's just a personal opinion, and not a strong enough one that it really rises to the level of needing to be a DASG-wide standard.

Double-however! I think it's useful -- for consistency's sake -- for DASG to have a single merge strategy that gets used across all the projects. Makes it easier for folks to play in other projects.

What are your thoughts?

Reply:
I'm a big fan of `Squash and Merge` as it keeps the commit history in `master` clean.  If digging is necessary, the PR from which code was added can be tracked down, and then the commit history can be viewed there.  Personal preference, I guess.

Reply:
Using squash and merge would also care for the concerns in the previous checkbox.

Reply:
@msnook, et al. I've tweaked the wording here to allow for programs that opt to use squash and merge. I want to mention _something_ along these lines still, as I think the squash and merge commit message starts off as a big combo of all the other commit messages, right?

Open to suggestions for better wording here!